### ISSUE TITLE
[On Hold] Upgrade nanoGPT

### ISSUE THREAD

**Issue by boris-drazic** [tenstorrent/tt-metal#3974]

**Comment by Sudharsan-V** [comment]
The tasks
```
 - store all intermedaite tensors in L1
 - store all weights in L1 (if they fit)
 
```
 are linked to [#4342](https://github.com/tenstorrent-metal/tt-metal/issues/4342)

**Comment by boris-drazic** [comment]
Can you make a PR for this?

**Comment by punithsekar** [comment]
We were planning to create a PR for ,
```
 - store all intermedaite tensors in L1
 - store all weights in L1 (if they fit)
```
after #4221 gets merged. As this work is depended on #4221 work.

Can we wait until #4221 gets merged or can we create a PR for it?

**Comment by boris-drazic** [comment]
Yes, you can wait.
